### PR TITLE
Publish Core and Relational functional tests (Resolves #2114)

### LINKED ISSUE #2114: Reuse of EntityFramework.Relational.FunctionalTests.dll
I have started writing a relational provider for EF7, and in order to run the test suite, I need the dll above. Maybe you should think about how to make it available to others. Just now, I have to build the dll and add a reference from my test project. Is there a better way (now or later) ? 
Looks like it will soon be the same issue with EF.Core.FunctionalTests.dll

### PR DESCRIPTION
**Pull request by bricelam**

### CODE REVIEW

**Comment by ErikEJ** [comment]
@bricelam  Great, but I also need: EntityFramework.Core.Tests.dll and EntityFramework.Relational.Tests.dll  :+1:

File: test/EntityFramework.Core.FunctionalTests/project.json [hunk 1]
```diff
@@ -1,20 +1,33 @@
...
+        "xunit.extensibility.core": "2.0.0-rc3-build2880"
+      },
+      "frameworkAssemblies": {
+        "System.Threading.Tasks": { "version": "4.0.0.0", "type": "build" },
+        "WindowsBase": "4.0.0.0"
+      }
+    },
     "dnx451": {
+      "dependencies": {
+        "xunit.runner.aspnet": "2.0.0-aspnet-*"
```
Comment (divega) [inline test/EntityFramework.Core.FunctionalTests/project.json]: Shouldn't we be using xunit.runner.dnx now?

Comment (bricelam) [inline test/EntityFramework.Core.FunctionalTests/project.json]: We haven't done the work to update yet.

**Comment by ajcvickers** [comment]
:shipit:

**Comment by bricelam** [comment]
@ErikEJ What do you need from the unit test assemblies? They're not extensible in the same way as the functional tests. Presumably, coping code from them into your own unit tests would be the better thing to do.

**Comment by ErikEJ** [comment]
@bricelam I currently need (and use):
From EntityFramework.Core.Tests:
EntityFrameworkServiceCollectionExtensionsTest
ApiConsistencyTestBase

From EntityFramework.Relational.Tests:
MigrationSqlGeneratorTestBase

**Pull request merged**